executor/set_test.go

```diff
@@ -518,16 +518,16 @@ func (s *testSerialSuite1) TestSetVar(c *C) {
 	tk.MustExec("SET GLOBAL tidb_enable_extended_stats = off")
 	tk.MustQuery("select @@global.tidb_enable_extended_stats").Check(testkit.Rows("0"))
 
-	tk.MustExec("SET SESSION tidb_enable_tiflash_fallback_tikv = on")
-	tk.MustQuery("select @@session.tidb_enable_tiflash_fallback_tikv").Check(testkit.Rows("1"))
-	tk.MustExec("SET SESSION tidb_enable_tiflash_fallback_tikv = off")
-	tk.MustQuery("select @@session.tidb_enable_tiflash_fallback_tikv").Check(testkit.Rows("0"))
-	tk.MustExec("SET GLOBAL tidb_enable_tiflash_fallback_tikv = on")
-	tk.MustQuery("select @@global.tidb_enable_tiflash_fallback_tikv").Check(testkit.Rows("1"))
-	tk.MustExec("SET GLOBAL tidb_enable_tiflash_fallback_tikv = off")
-	tk.MustQuery("select @@global.tidb_enable_tiflash_fallback_tikv").Check(testkit.Rows("0"))
-	c.Assert(tk.ExecToErr("SET SESSION tidb_enable_tiflash_fallback_tikv = 123"), NotNil)
-	c.Assert(tk.ExecToErr("SET GLOBAL tidb_enable_tiflash_fallback_tikv = 321"), NotNil)
+	tk.MustExec("SET SESSION tidb_allow_fallback_to_tikv = 'tiflash'")
+	tk.MustQuery("select @@session.tidb_allow_fallback_to_tikv").Check(testkit.Rows("tiflash"))
+	tk.MustExec("SET SESSION tidb_allow_fallback_to_tikv = ''")
+	tk.MustQuery("select @@session.tidb_allow_fallback_to_tikv").Check(testkit.Rows(""))
+	tk.MustExec("SET GLOBAL tidb_allow_fallback_to_tikv = 'tiflash'")
+	tk.MustQuery("select @@global.tidb_allow_fallback_to_tikv").Check(testkit.Rows("tiflash"))
+	tk.MustExec("SET GLOBAL tidb_allow_fallback_to_tikv = ''")
+	tk.MustQuery("select @@global.tidb_allow_fallback_to_tikv").Check(testkit.Rows(""))
+	c.Assert(tk.ExecToErr("SET SESSION tidb_allow_fallback_to_tikv = 'tikv,tiflash'"), NotNil)
```

Comment:
could you:

1. check the error message as well?
2. add test for
	- `set @@tidb_allow_fallback_to_tikv = 'tiflash, tiflash,   tiflash'`
	- `set @@tidb_allow_fallback_to_tikv = 'tidb, tiflash,   tiflash'`
	- `set @@tidb_allow_fallback_to_tikv = 'unknown, tiflash,   tiflash'`

Reply:
Is `set @@tidb_allow_fallback_to_tikv = 'tiflash, tiflash, tiflash'` legal or not? Now `set @@tidb_isolation_read_engines = 'tikv, tikv'` is legal but seems kind of wired.

Reply:
- `set @@tidb_allow_fallback_to_tikv = 'tiflash, tiflash, tiflash'` can be allowed, we can eliminate the duplicated values
- `@@tidb_isolation_read_engines = 'tikv, tikv'` can also be allowed, but if you decide to forbid this behavior, please report an error to the user to indicate the permitted values.

Reply:
also, please check the error message. checking whether the error is nil is insufficient